app/middleware/error-handler.js

```diff
@@ -1,8 +1,17 @@
 'use strict'
 
+const { NotFoundError } = require('../errors')
+const { response } = require('../utils/response')
+
 const logger = require('../utils/logger')(__filename)
 
 module.exports = function (err, req, res, next) {
+  if (err instanceof NotFoundError) {
+    logger.info(`NotFoundError handled: ${err.message}. Rendering 404 page`)
```

Comment:
We could probably add another story to improve the logging for products-ui to use the `AsyncLocalStorage` node feature to add useful context to all of the log lines - `product_external_id`, `gateway_account_id`, `x_request_id` etc.

For now until we do this, we should probably make sure we manually add the `x_request_id` to any log lines we add - we can get this from `req.correlationId`. This is so we can link together all log lines for a single request. For existing log lines, looks like we add this in square brackets at the start of the log line.

Reply:
Done - thanks